src/main/java/ch/njol/skript/expressions/ExprLastAttacker.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Projectile;
+import org.bukkit.event.entity.EntityDamageByBlockEvent;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.event.entity.EntityDamageEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.bukkitutil.ProjectileUtils;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+
+@Name("Last Attacker")
+@Description("The last block or entity that attacked an entity.")
+@Examples({"send \"%last attacker of event-entity%\""})
+@Since("INSERT VERSION")
+public class ExprLastAttacker extends SimplePropertyExpression<Entity, Object> {
+	
+	static {
+		register(ExprLastAttacker.class, Object.class, "last attacker", "entity");
+	}
+	
+	@Nullable
+	@Override
+	@SuppressWarnings("null")
+	public Object convert(Entity entity) {
+		EntityDamageEvent damageEvent = entity.getLastDamageCause();
```

Comment:
instead of duplicating this logic, can we just reuse the logic from ExprAttacker? you can do something like this:
```
init() {
    this.attackerExpr = new ExprAttacker();
    attackerExpr.init(exprs, 0, isDelayed, null);
}

convert(entity) {
    return attackerExpr.get(e.getLastDamageCause())[0];
}
```
that way we can ensure that both expressions always give the same attacker